examples/contract.kl

```diff
@@ -0,0 +1,78 @@
+#lang "n-ary-app.kl"
+[import "defun.kl"]
+[import "bool.kl"]
+[import [shift "n-ary-app.kl" 1]]
+[import [shift "defun.kl"1]]
+
+(meta
+  (defun contract-violation (contract)
+    (list-syntax ('error (quote '"Contract violation!")) contract)))
+
+(meta -- type annotation
```

Comment:
Ewww - I hope we get #60 done soon!

Reply:
Indeed! It would also be nice to have a `defun/ann` form like `defun/contract` where the contracts are replaced by type annotations.